website/source/api/secret/ad/index.html.md

```diff
@@ -0,0 +1,174 @@
+---
+
+# Active Directory Secrets Engine (API)
+
+This is the API documentation for the Vault AD secrets engine. For general
+information about the usage and operation of the AD secrets engine, please see
+the [Vault Active Directory documentation](/docs/secrets/ad/index.html).
+
+This documentation assumes the AD secrets engine is enabled at the `/ad` path
+in Vault. Since it is possible to enable secrets engines at any location, please
+update your API calls accordingly.
+
+## Configuration
+
+The `config` endpoint configures the LDAP connection and binding parameters, as well as the password rotation configuration.
+
+### Password parameters
+
+* `ttl` (string, optional) - The default password time-to-live in seconds. Once the ttl has passed, a password will be rotated the next time it's requested.
+* `max_ttl` (string, optional) - The maximum password time-to-live in seconds. No role will be allowed to set a custom ttl greater than the `max_ttl`.
+* `password_length` (string, optional) - The desired password length. Defaults to 64. Minimum is 14. Note: to meet complexity requirements, all passwords begin with "?@09AZ".
+
+### Connection parameters
+
+* `url` (string, required) - The LDAP server to connect to. Examples: `ldap://ldap.myorg.com`, `ldaps://ldap.myorg.com:636`. This can also be a comma-delineated list of URLs, e.g. `ldap://ldap.myorg.com,ldaps://ldap.myorg.com:636`, in which case the servers will be tried in-order if there are errors during the connection process.
+* `starttls` (bool, optional) - If true, issues a `StartTLS` command after establishing an unencrypted connection.
+* `insecure_tls` - (bool, optional) - If true, skips LDAP server SSL certificate verification - insecure, use with caution!
+* `certificate` - (string, optional) - CA certificate to use when verifying LDAP server certificate, must be x509 PEM encoded.
+
+### Binding parameters
+
+* `binddn` (string, required) - Distinguished name of object to bind when performing user and group search. Example: `cn=vault,ou=Users,dc=example,dc=com`
+* `bindpass` (string, required) - Password to use along with `binddn` when performing user search.
+* `userdn` (string, optional) - Base DN under which to perform user search. Example: `ou=Users,dc=example,dc=com`
+* `upndomain` (string, optional) - userPrincipalDomain used to construct the UPN string for the authenticating user. The constructed UPN will appear as `[username]@UPNDomain`. Example: `example.com`, which will cause vault to bind as `username@example.com`.
+
+## Config management
+
+At present, this endpoint does not confirm that the provided AD credentials are
+valid AD credentials with proper permissions.
+
+| Method   | Path                   | Produces               |
+| :------- | :--------------------- | :--------------------- |
+| `POST`   | `/ad/config`           | `204 (empty body)`     |
+| `GET`    | `/ad/config`           | `200 (see body below)` |
```

Comment:
We tend to be inconsistent about docs in general but I think most other backends have this as `200 application/json`

Reply:
:+1: